crates/net/eth-wire-types/src/primitives.rs

```diff
@@ -0,0 +1,83 @@
+//! Abstraction over primitive types in network messages.
+
+use std::fmt::Debug;
+
+use alloy_rlp::{Decodable, Encodable};
+
+/// Abstraction over primitive types which might appear in network messages. See
+/// [`crate::EthMessage`] for more context.
+pub trait NetworkPrimitives:
+    Send + Sync + Unpin + Clone + Debug + PartialEq + Eq + 'static
+{
+    /// The block header type.
+    type BlockHeader: Encodable
+        + Decodable
+        + Send
+        + Sync
+        + Unpin
+        + Clone
+        + Debug
+        + PartialEq
+        + Eq
+        + 'static;
+    /// The block body type.
+    type BlockBody: Encodable
+        + Decodable
+        + Send
+        + Sync
+        + Unpin
+        + Clone
+        + Debug
+        + PartialEq
+        + Eq
+        + 'static;
+    /// Full block type.
+    type Block: Encodable
+        + Decodable
+        + Send
+        + Sync
+        + Unpin
+        + Clone
+        + Debug
+        + PartialEq
+        + Eq
+        + 'static;
+
+    /// The transaction type which peers announce in `Transactions` messages. It is different from
+    /// `PooledTransactions` to account for Ethereum case where EIP-4844 transactions are not being
+    /// announced and can only be explicitly requested from peers.
+    type BroadcastedTransaction: Encodable
+        + Decodable
+        + Send
+        + Sync
+        + Unpin
+        + Clone
+        + Debug
+        + PartialEq
+        + Eq
+        + 'static;
+    /// The transaction type which peers return in `PooledTransactions` messages.
+    type PooledTransaction: Encodable
+        + Decodable
+        + Send
+        + Sync
+        + Unpin
+        + Clone
+        + Debug
+        + PartialEq
+        + Eq
+        + 'static;
+}
+
+/// Primitive types used by Ethereum network.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Hash)]
+#[non_exhaustive]
+pub struct EthNetworkPrimitives;
+
+impl NetworkPrimitives for EthNetworkPrimitives {
+    type BlockHeader = reth_primitives::Header;
+    type BlockBody = reth_primitives::BlockBody;
+    type Block = reth_primitives::Block;
+    type BroadcastedTransaction = reth_primitives::TransactionSigned;
+    type PooledTransaction = reth_primitives::PooledTransactionsElement;
+}
```

Comment:
this is basically double up of `NodePrimitives`, why the redundancy?

Reply:
it is very similar and we could perhaps try to unify them but I feel like we should keep them separate for now because that makes it easier to transition the network component